### DISCUSSION TITLE
allowing for function in rate_dropout as input brm_simulate_outline

### DISCUSSION THREAD

**Discussion by yonicd** [openpharma/brms.mmrm#103] [Ideas]
Would it make sense to allow for the users to input a function that would generate dropout shapes instead of assuming it is a scalar rate?

I think that most times the user has some prior parametric knowledge/data that they are basing the hypothetical on.

Another common scenario that may present itself is the user has some prior on dropout rate differences based on (sub)group, allowing for a vector of rates (instead of a single scalar) could also be a simpler implementation to put in.

**Comment by wlandau** [comment]
Interesting. If we were to pursue this, one option is to deprecate the `rate_dropout` argument in `brm_simulate_outline()`, then  explore a tidyverse-style layering system for dropouts. Something like:

```r
brm_simulate_outline() %>%
  brm_simulate_dropout_subgroup(...) %>%
  brm_simulate_continuous(...)
```

Or, we could encourage users to manually modify the `missing` column in the simulated data to suit their needs.

But I am not sure a highly nuanced approach for dropout simulation would belong in the scope of an MMRM-focused package. A constant dropout rate is very common and easy to simulate, and maybe a separate fit-for-purpose package would be more equipped to simulate more complicated kinds of missingness.
